Code/Engine/RendererCore/Pipeline/Implementation/Passes/BlendPass.cpp

```diff
@@ -0,0 +1,114 @@
+#include <RendererCore/RendererCorePCH.h>
+
+#include <Foundation/IO/TypeVersionContext.h>
+#include <RendererCore/Pipeline/Passes/BlendPass.h>
+#include <RendererCore/Pipeline/View.h>
+#include <RendererCore/RenderContext/RenderContext.h>
+
+#include <Core/Graphics/Geometry.h>
+#include <RendererCore/../../../Data/Base/Shaders/Pipeline/BlendConstants.h>
+
+// clang-format off
+EZ_BEGIN_DYNAMIC_REFLECTED_TYPE(ezBlendPass, 1, ezRTTIDefaultAllocator<ezBlendPass>)
+{
+  EZ_BEGIN_PROPERTIES
+  {
+    EZ_MEMBER_PROPERTY("InputA", m_PinInputA),
+    EZ_MEMBER_PROPERTY("InputB", m_PinInputB),
+    EZ_MEMBER_PROPERTY("Output", m_PinOutput),
+    EZ_MEMBER_PROPERTY("BlendFactor", m_fBlendFactor)->AddAttributes(new ezDefaultValueAttribute(0.5f)),
+  }
+  EZ_END_PROPERTIES;
+}
+EZ_END_DYNAMIC_REFLECTED_TYPE;
+// clang-format on
+
+ezBlendPass::ezBlendPass()
+  : ezRenderPipelinePass("BlendPass")
+{
+  m_hShader = ezResourceManager::LoadResource<ezShaderResource>("Shaders/Pipeline/Blend.ezShader");
+  EZ_ASSERT_DEV(m_hShader.IsValid(), "Could not load blend shader!");
+  m_hBlendCB = ezRenderContext::CreateConstantBufferStorage<ezBlendConstants>();
+}
+
+ezBlendPass::~ezBlendPass()
+{
+  ezRenderContext::DeleteConstantBufferStorage(m_hBlendCB);
+  m_hBlendCB.Invalidate();
+}
+
+bool ezBlendPass::GetRenderTargetDescriptions(const ezView& view, const ezArrayPtr<ezGALTextureCreationDescription* const> inputs, ezArrayPtr<ezGALTextureCreationDescription> outputs)
+{
+  // Color
+  if (inputs[m_PinInputA.m_uiInputIndex] && inputs[m_PinInputB.m_uiInputIndex])
+  {
+    if (!inputs[m_PinInputA.m_uiInputIndex]->m_bAllowShaderResourceView || !inputs[m_PinInputB.m_uiInputIndex]->m_bAllowShaderResourceView)
+    {
+      ezLog::Error("Blend pass inputs must allow shader resource view.");
+      return false;
+    }
+
+    outputs[m_PinOutput.m_uiOutputIndex] = *inputs[m_PinInputA.m_uiInputIndex];
+  }
+  else
+  {
+    ezLog::Error("No input connected to blend pass!");
+    return false;
+  }
+
+  return true;
+}
+
+void ezBlendPass::Execute(const ezRenderViewContext& renderViewContext, const ezArrayPtr<ezRenderPipelinePassConnection* const> inputs, const ezArrayPtr<ezRenderPipelinePassConnection* const> outputs)
+{
+  if (outputs[m_PinOutput.m_uiOutputIndex])
+  {
+    ezGALDevice* pDevice = ezGALDevice::GetDefaultDevice();
+
+    ezBlendConstants* cb = ezRenderContext::GetConstantBufferData<ezBlendConstants>(m_hBlendCB);
```

Comment:
Would be a nice use case for push constants

Reply:
Good point! Fixed!